.github/workflows/ci.yml

```diff
@@ -45,4 +110,263 @@ jobs:
         uses: codecov/codecov-action@v2
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
-          file: coverage.txt
+          flags: server
+
+  slack-notification:
+    if: success() || failure()
+    name: Slack Notification
+    needs: [ci-web, ci-server]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Slack Notification
+        uses: Gamesight/slack-workflow-status@master
+        if: always()
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          slack_webhook_url: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+  info:
+    name: Collect information
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion != 'failure' && github.event.repository.full_name == 'reearth/reearth' && (github.event.workflow_run.head_branch == 'release' || !startsWith(github.event.head_commit.message, 'v'))
+    outputs:
+      sha_short: ${{ steps.info.outputs.sha_short }}
+      tag: ${{ steps.info.outputs.tag }}
+      tag_short: ${{ steps.info.outputs.tag_short }}
+      name: ${{ steps.info.outputs.name }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.workflow_run.head_sha }}
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
+      - name: Get info
+        id: info
+        env:
+          BRANCH: ${{ github.event.workflow_run.head_branch }}
+        # The tag name should be retrieved lazily, as tagging may be delayed.
+        run: |
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+          if [[ "$BRANCH" = "release" ]]; then
+            TAG=$(git tag --points-at HEAD)
+            if [[ ! -z "$TAG" ]]; then
+              echo "::set-output name=tag::$TAG"
+              echo "::set-output name=tag_short::${TAG#v}"
+            else
+              echo "::set-output name=name::rc"
+            fi
+          else
+            echo "::set-output name=name::nightly"
+          fi
+      - name: Show info
+        env:
+          SHA_SHORT: ${{ steps.info.outputs.sha_short }}
+          TAG: ${{ steps.info.outputs.tag }}
+          TAG_SHORT: ${{ steps.info.outputs.tag_short }}
+          NAME: ${{ steps.info.outputs.name }}
+        run: echo "sha_short=$SHA_SHORT, tag=$TAG, tag_short=$TAG_SHORT, name=$NAME"
+
+  build-server:
+    name: Build and release server
+    runs-on: ubuntu-latest
+    needs: [ci-web, ci-server, info]
```

Comment:
Why do we need to do `build-server` depends on `ci-web`?

Reply:
This was added to make sure that `build-server` is not executed if `ci-web` failed.

Reply:
```suggestion
    needs: info
```

is not working?

Reply:
It will work but we'd want to build-sever only when `ci-server` has passed and `ci-web` is either skipped or was success 
https://github.com/reearth/reearth/blob/958437fede8a4eb2cc1d97403301fbd9a42fcfa2/.github/workflows/ci.yml#L162

Reply:
hm ok